### PR TITLE
feat(logs): new pipeline design

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

[> Link to the JIRA ticket
](https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-596)

- Add new UI for application and deployment logs
- Updated Cluster UI logs

Capture d’écran 2023-04-11 à 16 50 00

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [x] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by evoxmusic** [comment]
A preview environment was automatically created via Qovery.
Click on the link below to follow its deployment and use it.
:point_right: [[PR] staging - feat(logs): new pipeline design - 2023-03-27T09:06:52Z](https://console.qovery.com/organization/141c07c8-0dd9-4623-983b-3fdd61867255/project/977bd069-20f7-4b24-bd95-28b44174cf43/environment/63e01ae4-306d-49d5-b7e4-10bffdd7057a/services/general)

 Another comment will be posted when deployments are terminated

**Comment by evoxmusic** [comment]
Your preview environment has been successfully deployed !
Click on the link below to open your service:
:point_right: [console](https://z63e01ae4-z12af5d3e-gtw.za8ad0659.bool.sh)
:point_right: [storybook](https://z63e01ae4-z61bdf1e8-gtw.za8ad0659.bool.sh)

File: libs/pages/logs/environment/src/lib/page-environment-logs.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+          element={<PodLogsFeature clusterId={environment?.cluster_id || ''} setServiceId={setServiceId} />}
+        />
+      </Routes>
+      {(location.pathname === `${ENVIRONMENT_LOGS_URL(organizationId, projectId, environmentId)}/` ||
+        location.pathname === ENVIRONMENT_LOGS_URL(organizationId, projectId, environmentId)) && (
+        <div className="flex justify-center w-[calc(100%-8px)] min-h-full bg-element-light-darker-400 m-1 rounded">
+          <div className="flex flex-col items-center mt-12">
+            <Icon name={IconAwesomeEnum.WRENCH} className="text-text-300" />
+            <p className="text-text-300 font-medium">
+              Environment is {environmentStatus?.state.toLowerCase().replace('_', ' ')}, please select a service on the
```
Comment (acarranoqovery) [inline libs/pages/logs/environment/src/lib/page-environment-logs.tsx:73]: "Please select a service on the left menu to access its deployment logs or live logs". We can stay generic enough here and instead have the real status on the left panel

File: libs/shared/console-shared/src/lib/layout-logs/layout-logs.tsx [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+          {data?.loadingStatus === 'not loaded' || !data ? (
+            <LoaderSpinner className="w-6 h-6" theme="dark" />
+          ) : (
+            <div className="flex flex-col items-center">
+              <img
+                className="w-40 pointer-events-none user-none"
+                src="/assets/images/event-placeholder-dark.svg"
+                alt="Event placeholder"
+              />
+              <p className="mt-5 text-text-100 font-medium text-center">{placeholderDescription}</p>
```
Comment (acarranoqovery) [inline libs/shared/console-shared/src/lib/layout-logs/layout-logs.tsx:129]: If possible,  we should take into account these conditions:
    - if depl history empty (environment never deployed before): "This service has never been deployed and no thus logs are available."
    - else (service deployed at least once but not this time)  "This service is not being deployed right now and thus no deployment logs are available. You can access the application logs from the "Live Logs" tab". Not sure if we can add a button to open the tab

File: libs/shared/utils/src/lib/tools/merge-deployment-services.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import { DeploymentHistoryEnvironment, JobScheduleEvent } from 'qovery-typescript-axios'
+import { ServiceTypeEnum } from '@qovery/shared/enums'
+import { DeploymentService } from '@qovery/shared/interfaces'
+
+export const mergeDeploymentServices = (deploymentHistory?: DeploymentHistoryEnvironment[]) => {
```
Comment (bdebon) [inline libs/shared/utils/src/lib/tools/merge-deployment-services.ts:5]: This file could have a small test

File: libs/pages/logs/environment/src/lib/feature/deployment-logs-feature/deployment-logs-feature.tsx [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        setLoadingStatus('loaded')
+
+        const newLog = JSON.parse(message?.data)
+
+        if (pauseStatusLogs) {
+          setPauseLogs((prev: EnvironmentLogs[]) => [...prev, ...newLog])
+        } else {
+          setLogs((prev: EnvironmentLogs[]) => {
+            // return unique log by timestamp
+            return [...new Map([...prev, ...pauseLogs, ...newLog].map((item) => [item['timestamp'], item])).values()]
```
Comment (bdebon) [inline libs/pages/logs/environment/src/lib/feature/deployment-logs-feature/deployment-logs-feature.tsx:80]: 😮

**Comment by RemiBonnet** [comment]
> It should be nice that the log remembers from where it has been opened so that we can go back to the correct page when we click on close. Right now, if you open the logs from the services list, it's fine because closing them bring you back to the same page. If you open the log from the service detail, then if you close, you're brought back to the services list page when it would be less confusing that it brings you back to the service list. Wdyt? cc @alecarrano

I agree, I'll merge this PR before and add other improvements like this for the next time

**Comment by evoxmusic** [comment]
Youhou 🔥 Well done @RemiBonnet

**Pull request merged**